File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
// n: 벽 길이
// m: 롤러의 길이
// section: 다시 칠해야하는 구역

function solution(n, m, section) {
    let answer = 0;
    let painted = new Array(n+1).fill(0);
    
    for (s of section) {
        if (painted[s]) continue;
        for (let i=s; i<s+m; i++) {
            painted[i] = 1;
        }
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        boolean[] completes = new boolean[n+1];
        for (int i=0; i<section.length; i++) {
            if (completes[section[i]]) continue;
            for (int j=section[i]; j<section[i] + m; j++) {
                if (j > n) break; 
                completes[j] = true;
            }
            answer++;
        }
        return answer;
    }
}